test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs

```diff
@@ -2693,6 +2696,107 @@ public async Task DotnetRestore_WithMultiTargettingProject_WhenTargetFrameworkIs
             allTargets.Should().Contain(condition);
         }
 
+        // https://github.com/NuGet/Home/issues/13829
+        [PlatformFact(SkipPlatform = Platform.Linux)]
+        public async Task DotnetRestore_WithServerProvidingAuditData_RestoreTaskReturnsCountProperties()
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            using var mockServer = new FileSystemBackedV3MockServer(pathContext.PackageSource, sourceReportsVulnerabilities: true);
+
+            mockServer.Vulnerabilities.Add(
+                "Insecure.Package",
+                new List<(Uri, PackageVulnerabilitySeverity, VersionRange)> {
+                    (new Uri("https://contoso.test/advisories/12346"), PackageVulnerabilitySeverity.Critical, VersionRange.Parse("[1.0.0, 2.0.0)"))
+                });
+            pathContext.Settings.RemoveSource("source");
+            pathContext.Settings.AddSource("source", mockServer.ServiceIndexUri, allowInsecureConnectionsValue: "true");
+
+            var packageA1 = new SimpleTestPackageContext()
+            {
+                Id = "packageA",
+                Version = "1.0.0"
+            };
+
+            await SimpleTestPackageUtility.CreatePackagesAsync(
+                pathContext.PackageSource,
+                packageA1);
+
+            var targetFramework = Constants.DefaultTargetFramework.GetShortFolderName();
+            string csprojContents = GetProjectFileForRestoreTaskOutputTests(targetFramework);
+            var csprojPath = Path.Combine(pathContext.SolutionRoot, "test.csproj");
+            File.WriteAllText(csprojPath, csprojContents);
+
+            mockServer.Start();
+
+            // Act & Assert
+
+            // First restore is fresh, so should not be skipped (no-op), and should run audit
+            _dotnetFixture.RunDotnetExpectSuccess(pathContext.SolutionRoot, "restore -p:ExpectedRestoreProjectCount=1 -p:ExpectedRestoreSkippedCount=0 -p:ExpectedRestoreProjectsAuditedCount=1");
+
+            // Second restore is no-op, so should be skipped, and audit not run
+            _dotnetFixture.RunDotnetExpectSuccess(pathContext.SolutionRoot, "restore -p:ExpectedRestoreProjectCount=1 -p:ExpectedRestoreSkippedCount=1 -p:ExpectedRestoreProjectsAuditedCount=0");
+
+            mockServer.Stop();
+        }
+
+        [Fact]
+        public async Task DotnetRestore_WithServerWithoutAuditData_RestoreTaskReturnsCountProperties()
```

Comment:
I didn't follow this test name well, but reading the test, my interpretation is it's expecting 1 restore, then 1 no-op restore. In neither case is audit count expected to tick.

Reply:
Yes, you're right. It does two restores, the second is a no-op. No-op (should) never run NuGetAudit, and the first restore does not run audit because the server did not provide a vulnerability database ("audit data").

I couldn't think of a way to write this in a name without it being unreasonably long.